### PR TITLE
Adding text about PCEP

### PR DESCRIPTION
**Pull request by italobusi**
Adding text proposal from Dhruv on a new section 3.3 to address open issue #27

Co-authored-by: dhruvdhody <dhruv.ietf@gmail.com>

### CODE REVIEW

File: draft-ietf-teas-actn-poi-applicability.txt [hunk 1]
```diff
@@ -1014,6 +1014,30 @@ Internet-Draft                 ACTN POI                   November 2020
    [Editor's note:] Add YANG models used for tunnel and service
    configuration.
 
+3.3. PCEP
+
+   [RFC8623] examines the applicability of a Path Computation Element
```
Comment (dancecca) [inline draft-ietf-teas-actn-poi-applicability.txt:1019]: RFC 8623 is       Stateful Path Computation Element (PCE) Protocol Extensions  for Usage with Point-to-Multipoint TE Label Switched Paths (LSPs)
Maybe you meant RFC8637

File: draft-ietf-teas-actn-poi-applicability.txt [hunk 2]
```diff
@@ -1014,6 +1014,30 @@ Internet-Draft                 ACTN POI                   November 2020
...
+   (PCE) [RFC5440] and PCE Communication Protocol (PCEP) to the ACTN
+   framework. It further describes how the PCE architecture is
+   applicable to ACTN.  It also lists the PCEP extensions that are
```
Comment (dancecca) [inline draft-ietf-teas-actn-poi-applicability.txt:1022]: just to improve readability: It further describes how the PCE architecture is applicable to ACTN and lists the PCEP extensions...

File: draft-ietf-teas-actn-poi-applicability.txt [hunk 3]
```diff
@@ -1014,6 +1014,30 @@ Internet-Draft                 ACTN POI                   November 2020
...
+   ACTN.
+
+   Since the PCEP supports path computation in the packet as well as
+   optical networks, PCEP is well suited for inter-layer path
+   computation. [RFC5623] describes a framework for applying the PCE-
+   based architecture to interlayer (G)MPLS traffic engineering.
+   Further, the section 6.1 of [RFC8751] states the H-PCE applicability
+   for inter-layer or POI.
+
+   [RFC8623] lists various PCEP extensions that are applicable to ACTN.
```
Comment (dancecca) [inline draft-ietf-teas-actn-poi-applicability.txt:1036]: wrong reference also here?

File: draft-ietf-teas-actn-poi-applicability.txt [hunk 4]
```diff
@@ -1014,6 +1014,30 @@ Internet-Draft                 ACTN POI                   November 2020
...
+   It also list the PCEP extension for optical network and POI. Note
+   that the PCEP can be used in conjunction with the YANG models
+   described in the rest of this document.
+
```
Comment (dancecca) [inline draft-ietf-teas-actn-poi-applicability.txt:1040]: maybe we can spend few words to say how? it is true that you can you PCE for some things and YANG for others but what would be the benefit?

**Comment by jbouqui153** [comment]
Section 3.3 PCEP is fine from an informational perspective. however, from an operator perspective, I think we would need to explain that there are at least two main possible implementations of ACTN hierarchical architecture with respect to H-PCE and PCEs : 

1) Option 1:  Hierarchical Controller (MDSC) with H-PCE function embedded and using single RESTCONF/YANG interface towards each PNC (IP PNC or OP PNC or MW PNC). Over that interface MDSC discovers all the TE information required from PNCs through the use of the RFC8795 [te-topo] data model and can also request the creation of TE tunnels [yang-te]. It may either perform full multi-layer path computation or delegate path computation to the underneath PNC thanks to [yang-path-computation] data model. This approach is very attractive for operators from an multi-vendor integration perspective as it is simple and we need only one type of interface (RESTCONF) and use the relevant YANG data models depending on the operator use case considered. Benefits of having only one protocol for the MPI between MDSC and PNC have been already highlighted in  [yang-path-computation] draft. Single protocol and consistency of the global network/LSP/service database at Hierarchical controller level are the main ones in my opinion.

2) Option 2: similar to 1) but PCEP is added in addition to the RESTCONF/YANG interface to follow hierarchical PCE architecture as explained by Dhruv. As mentioned in Option 1, from an operator perspective it adds integration complexity to have two protocols instead of one. 

3) H-PCE function is outside the Hierarchical Controller (MDSC) (and PCE function may reside also outside of the PNCs). This is possible from an implementation perspective but introduces the need for PCEP on top of the RESTCONF protocol for the MPI. From an integration perspective this leads to a higher complexity as compared to RESTCONF only for the MPI and this may lead as well to inconsistencies issues between H-PCE and MDSC or PNCs and PCEs unless proper synchronization is implemented. Last but not least PCEP is widely supported by most IP PNC Vendors in SBI but not sure about the NBI support while for OP PNCs there are very few Vendors supporting PCEP even in SBI today…

So, from an implementation perspective, and in order to accelerate the SDN delivery required for overall automation of operators’ transport networks, I personally believe that an operator would rather chose option 1 vs options 2 & 3. 
 
Of course this is only my personal view and we can discuss this topic further  in our next calls in order to agree on final text.

File: draft-ietf-teas-actn-poi-applicability.txt (hunk 1 above)
Comment (italobusi) [reply draft-ietf-teas-actn-poi-applicability.txt:1019]: **2020-11-09 TEAS**

Correct: it should be RFC8637

File: draft-ietf-teas-actn-poi-applicability.txt (hunk 2 above)
Comment (italobusi) [reply draft-ietf-teas-actn-poi-applicability.txt:1022]: **2020-11-09 TEAS**

Ok

File: draft-ietf-teas-actn-poi-applicability.txt (hunk 3 above)
Comment (italobusi) [reply draft-ietf-teas-actn-poi-applicability.txt:1036]: **2020-11-09 TEAS**

Correct: it should be RFC8637

File: draft-ietf-teas-actn-poi-applicability.txt (hunk 4 above)
Comment (italobusi) [reply draft-ietf-teas-actn-poi-applicability.txt:1040]: **2020-11-09 TEAS**

ACTN can support multiple options at the MPI: it is possible to use only PCEP, as defined in RFC8637, or only the YANG models, as described in this document, or any possible combinations of PCEP and YANG models.

The scope of the text is just to clarify that all these possibilities exist.

For example, it is possible to use PCEP just for path computation and YANG models for the rest  (for example when PCEP has been already implemented/deployed).

@dancecca : are you ok to just add an example?

@sergiobelotti : please check if this example is reported in some public document

**Comment by italobusi** [comment]
> Section 3.3 PCEP is fine from an informational perspective. however, from an operator perspective, I think we would need to explain that there are at least two main possible implementations of ACTN hierarchical architecture with respect to H-PCE and PCEs :
> 
> 1. Option 1:  Hierarchical Controller (MDSC) with H-PCE function embedded and using single RESTCONF/YANG interface towards each PNC (IP PNC or OP PNC or MW PNC). Over that interface MDSC discovers all the TE information required from PNCs through the use of the RFC8795 [te-topo] data model and can also request the creation of TE tunnels [yang-te]. It may either perform full multi-layer path computation or delegate path computation to the underneath PNC thanks to [yang-path-computation] data model. This approach is very attractive for operators from an multi-vendor integration perspective as it is simple and we need only one type of interface (RESTCONF) and use the relevant YANG data models depending on the operator use case considered. Benefits of having only one protocol for the MPI between MDSC and PNC have been already highlighted in  [yang-path-computation] draft. Single protocol and consistency of the global network/LSP/service database at Hierarchical controller level are the main ones in my opinion.
> 2. Option 2: similar to 1) but PCEP is added in addition to the RESTCONF/YANG interface to follow hierarchical PCE architecture as explained by Dhruv. As mentioned in Option 1, from an operator perspective it adds integration complexity to have two protocols instead of one.
> 3. H-PCE function is outside the Hierarchical Controller (MDSC) (and PCE function may reside also outside of the PNCs). This is possible from an implementation perspective but introduces the need for PCEP on top of the RESTCONF protocol for the MPI. From an integration perspective this leads to a higher complexity as compared to RESTCONF only for the MPI and this may lead as well to inconsistencies issues between H-PCE and MDSC or PNCs and PCEs unless proper synchronization is implemented. Last but not least PCEP is widely supported by most IP PNC Vendors in SBI but not sure about the NBI support while for OP PNCs there are very few Vendors supporting PCEP even in SBI today…
> 
> So, from an implementation perspective, and in order to accelerate the SDN delivery required for overall automation of operators’ transport networks, I personally believe that an operator would rather chose option 1 vs options 2 & 3.
> 
> Of course this is only my personal view and we can discuss this topic further in our next calls in order to agree on final text.

**2020-11-09 TEAS**

The objective of the text is to describe that there are other possibilities without precluding option 1 (i.e., just use RESTCONF/YANG).

There are some concerns with option 3 since a PCE function is anyway required within MDSC even if PCEP is not used. The option to have a PCE function outside of the controller is not described in ACTN framework nor in the previous section.

Other options are possible (e.g., when adding RESTCONF/YANG on an already deployed PCEP implementation).

One possibility is to improve the description of these options (either somewhere in section 3 or at the beginning of section 4), taking into account the feedbacks from this call, and indicate that section 4 of the draft focuses on analyzing option 1.

@jbouqui153 : what do you think?

**Review by jbouqui153 — APPROVED** [review]

**Review by sergiobelotti — APPROVED** [review]

**Review by FabioPeruzzini — APPROVED** [review]

**Review by jbouqui153 — APPROVED** [review]
Sorry for the delay. I am fine with the changes proposed

**Pull request merged**